client/http/client.go

```diff
@@ -191,20 +192,31 @@ func (c *client) execDuration(name string, duration time.Duration) {
 	c.executionDuration.WithLabelValues(name).Observe(duration.Seconds())
 }
 
+// HeaderOption configures the HTTP header.
+type HeaderOption func(header http.Header)
+
+// WithAllowFollowerHandle sets the header field to allow a PD follower to handle this request.
+func WithAllowFollowerHandle(allow bool) HeaderOption {
+	return func(header http.Header) {
+		header.Set("PD-Allow-follower-handle", fmt.Sprintf("%t", allow))
```

Comment:
It's better to format it as `PD-Allow-Follower-Handle`. Also, `CanonicalMIMEHeaderKey` has helped us do this, but I think it's clear if we do this.